src/main/java/org/wso2/carbon/apimgt/securityenforcer/publisher/sync/SyncPublisher.java

```diff
@@ -100,17 +100,17 @@ public AseResponseDTO publishSyncEvent(JSONObject requestBody, String correlatio
     public boolean verifyRequest(JSONObject requestMetaData, String requestCorrelationID) throws AISecurityException {
         AseResponseDTO aseResponseDTO = publishSyncEvent(requestMetaData, requestCorrelationID,
                 AISecurityHandlerConstants.ASE_RESOURCE_REQUEST);
-        if (AISecurityHandlerConstants.ASE_RESPONSE_CODE_SUCCESS == aseResponseDTO.getResponseCode()) {
-            if (log.isDebugEnabled()) {
-                log.debug("Access granted by the Ping AI handler for the request id " + requestCorrelationID);
-            }
-            return true;
-        } else {
+        if (AISecurityHandlerConstants.ASE_RESPONSE_CODE_FORBIDDEN == aseResponseDTO.getResponseCode()) {
```

Comment:
This condition has been switched. But **why** is not documented in code comments or commit message. These is very important to document when debugging issues later. Commit message is the ideal place for this documentation but now we don't have that option. So shall we add a code comment?

Reply:
Fixed with https://github.com/wso2-extensions/apim-handler-pingai/pull/22